### DISCUSSION TITLE
Custom Middleware for CRM boilerplate

### DISCUSSION THREAD

**Discussion by ghost** [refinedev/refine#5455] [Q&A] [answered]
Hello,
I wanna integrate my custom middleware into my new CRM boilerplate. But now I wanna observe the existing REST responses in Swagger or like this API testing tool or something to observe the APIs and the DTOs used. Based on them I wanna create my middleware. How I can do that?
Thanks.

**Comment by BatuhanW** [comment, marked as answer]
Hello @aislam1234 we aren't using REST for CRM, but we are using GraphQL. You can find the full API documentation here: https://api.crm.refine.dev/graphql

**Reply by ghost** [reply]
thanks @BatuhanW , let me check this out.
